programs/mmm/src/instructions/ext_vanilla/ext_fulfill_buy.rs

```diff
@@ -0,0 +1,356 @@
+        try_close_escrow, try_close_sell_state,
+    },
+    SolFulfillBuyArgs,
+};
+
+// ExtSolFulfillBuy means a seller wants to sell NFT into the pool
+// where the pool has some buyside payment liquidity. Therefore,
+// the seller expects a min_payment_amount that goes back to the
+// seller's wallet for the asset_amount that the seller wants to sell.
+// This is mainly used for Token22 extension
+#[derive(Accounts)]
+#[instruction(args:SolFulfillBuyArgs)]
+pub struct ExtSolFulfillBuy<'info> {
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    /// CHECK: we will check the owner field that matches the pool owner
+    #[account(mut)]
+    pub owner: UncheckedAccount<'info>,
+    pub cosigner: Signer<'info>,
+    #[account(mut)]
+    /// CHECK: we will check that the referral matches the pool's referral
+    pub referral: UncheckedAccount<'info>,
+    #[account(
+        mut,
+        seeds = [POOL_PREFIX.as_bytes(), owner.key().as_ref(), pool.uuid.as_ref()],
+        has_one = owner @ MMMErrorCode::InvalidOwner,
+        has_one = referral @ MMMErrorCode::InvalidReferral,
+        has_one = cosigner @ MMMErrorCode::InvalidCosigner,
+        constraint = pool.payment_mint.eq(&Pubkey::default()) @ MMMErrorCode::InvalidPaymentMint,
+        constraint = pool.expiry == 0 || pool.expiry > Clock::get().unwrap().unix_timestamp @ MMMErrorCode::Expired,
+        bump
+    )]
+    pub pool: Box<Account<'info, Pool>>,
+    /// CHECK: it's a pda, and the private key is owned by the seeds
+    #[account(
+        mut,
+        seeds = [BUYSIDE_SOL_ESCROW_ACCOUNT_PREFIX.as_bytes(), pool.key().as_ref()],
+        bump,
+    )]
+    pub buyside_sol_escrow_account: UncheckedAccount<'info>,
+    /// CHECK: check_allowlists_for_mint_ext
+    pub asset_mint: Box<InterfaceAccount<'info, Mint>>,
+    #[account(
+        mut,
+        token::mint = asset_mint,
+        token::authority = payer,
+        token::token_program = token_program,
+    )]
+    pub payer_asset_account: Box<InterfaceAccount<'info, TokenAccount>>,
+    /// CHECK: check in init_if_needed_ata
+    #[account(mut)]
+    pub sellside_escrow_token_account: UncheckedAccount<'info>,
+    /// CHECK: check in init_if_needed_ata
+    #[account(mut)]
+    pub owner_token_account: UncheckedAccount<'info>,
+    /// CHECK: will be used for allowlist checks
+    pub allowlist_aux_account: UncheckedAccount<'info>,
+    #[account(
+        init_if_needed,
+        payer = payer,
+        seeds = [
+            SELL_STATE_PREFIX.as_bytes(),
+            pool.key().as_ref(),
+            asset_mint.key().as_ref(),
+        ],
+        space = SellState::LEN,
+        bump
+    )]
+    pub sell_state: Account<'info, SellState>,
+    pub system_program: Program<'info, System>,
+    pub token_program: Interface<'info, TokenInterface>,
+    pub associated_token_program: Program<'info, AssociatedToken>,
+    pub rent: Sysvar<'info, Rent>,
```

Comment:
is rent required?

Reply:
yeah it is required for the `init_if_needed_at`

Reply:
I think should be good to get rid of rent account now; in other places the init_if_needed doesn't require this account anymore